packages/eslint-plugin/tests/rules/no-duplicate-type-constituents.test.ts

```diff
@@ -673,5 +674,21 @@ type T = Record<string, A  >;
         },
       ],
     },
+    {
+      code: '(a?: string | undefined) => {};',
```

Comment:
[Testing] Could you add some test cases with `FunctionDeclaration`, `FunctionExpression` and node types mentioned in my comment above?